components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/SubscriptionDeletionWorkflow.java

```
@@ -75,8 +76,12 @@ public WorkflowResponse completeWorkflow(WorkflowExecutor workflowExecutor) thro
             if (log.isDebugEnabled()) {
                 log.debug("Subscription deletion workflow complete: Approved");
             }
-            apiGateway.deleteAPISubscription(subscription);
             apiSubscriptionDAO.deleteAPISubscription(getWorkflowReference());
+            try {
+                apiGateway.deleteAPISubscription(subscription);
```

don;t we need to throw the error upwards?